src/web/mjs/connectors/templates/Lezhin.mjs

```diff
@@ -68,165 +92,153 @@ export default class Lezhin extends Connector {
-            20 |  640w |  720w
-            30 |  720w | 1080w
-            40 | 1080w | 1280w
+        q  | Free  | Purchased
+        ----------------------
+        10 |  480w |  640w
+        20 |  640w |  720w
+        30 |  720w | 1080w
+        40 | 1080w | 1280w
         */
-        const script = `
-            new Promise((resolve, reject) => {
-                setTimeout(() => {
-                    try {
-                        if(window.location.pathname.includes('/login')) {
-                            throw new Error('You need to be logged in to access the content of this chapter!');
-                        }
-                        const subscribed = __LZ_DATA__.product && __LZ_DATA__.product.subscribed;
-                        const purchased = __LZ_DATA__.purchased && __LZ_DATA__.purchased.includes(__LZ_DATA__.episode.id);
-                        const hasOnlyJPEG = !!__LZ_DATA__.episode.d2;
-                        const images = __LZ_DATA__.episode.scrollsInfo.map(x => {
-                            const extension = ${this.config.forceJPEG.value} || hasOnlyJPEG ? '.jpg' : '.webp';
-                            const uri = new URL('/v2' + x.path + extension, __LZ_CONFIG__.cdnUrl);
-                            uri.searchParams.set('access_token', '${this.accessToken}');
-                            uri.searchParams.set('purchased', subscribed || purchased);
-                            uri.searchParams.set('q', 40);
-                            return uri.href;
-                        });
-                        resolve(images);
-                    } catch(error) {
-                        reject(error);
-                    }
-                }, 2500);
-            });
-        `;
 
-        const uri = new URL(`${this.url}/comic/${chapter.manga.id}/${chapter.id}`);
-        const request = new Request(uri, this.requestOptions);
-        return Engine.Request.fetchUI(request, script);
+        //https://www.lezhin.com/lz-api/v2/inventory_groups/comic_viewer?platform=web&store=web&alias=angel&name=1&preload=false&type=comic_episode
+        let uri = new URL('https://www.lezhin.com/lz-api/v2/inventory_groups/comic_viewer');
+        uri.searchParams.set('platform', 'web');
+        uri.searchParams.set('store', 'web');
+        uri.searchParams.set('alias', chapter.manga.id);
+        uri.searchParams.set('name', chapter.id);
+        uri.searchParams.set('preload', false);
+        uri.searchParams.set('type', 'comic_episode');
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchJSON(request);
+
+        const episode = data.data.extra.episode;
+        let purchased = episode.coin == 0;
+        //purchased = purchased || (episode.freedAt && episode.freedAt < Date.now());
+        const subscribed = data.data.extra.subscribed;
+        const updatedAt = episode.updatedAt;
+
+        let pictures = [];
+
+        for (let i = 0; i < episode.scrollsInfo.length; i++) {
```

Comment:
Get pictures is supposed to be fast, but now it makes a request for each image which may take a long time.
I recommend to leverage the `_handleConnectorURI` mechanism for requesting the image data and make these requests there.